File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//品牌服务层
app.service("brandService",function($http){

    //查询所有
    this.findAll = function(){
        return $http.get("../brand/findAll.do");
    }

    //分页查询_服务层
    this.findPage = function(pageNum,pageSize){
        return $http.get("../brand/findPage.do?pageNum=" + pageNum +
                                               "&pageSize=" + pageSize);
    }

    //根据ID查询_服务层
    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    }

    //新增品牌_服务层
    this.add = function(entity){
        return $http.post("../brand/add.do",entity)
    }

    //修改品牌_服务层
    this.update = function(entity){
        return $http.post("../brand/update.do",entity)
    }

    //批量删除
    this.del = function(ids){
        return  $http.get("../brand/del.do?ids="+ids)
    }

    //条件查询
    this.search = function(pageNum,pageSize,searchEntity){
        return $http.post("../brand/search.do?pageNum=" + pageNum +
            "&pageSize=" + pageSize,searchEntity)
    }

    //查询所有品牌显示在下拉框
    this.selectOptionList = function(){
        return $http.get("../brand/selectOptionList.do");
    }
})

File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.github.pagehelper.PageInfo;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

/**
 * 品牌服务接口
 */
public interface BrandService {

    public List<TbBrand> findAll();

    PageInfo findPage(TbBrand brand, Integer pageNum, Integer pageSize);

    PageInfo findPage(int pageNum, int pageSize);

    void add(TbBrand brand);

    TbBrand findOne(long id);

    void update(TbBrand brand);

    void del(Long[] ids);

    List<Map> selectOptionList();
}
